### PR TITLE
TEP-0059: Remove `scope-when-expressions-to-task` feature flag

### PR DESCRIPTION
**Pull request by jerop**
# Changes

In [TEP-0007: Conditions Beta][tep-0007], we introduced `when` expressions to guard execution of `Tasks` in `Pipelines`. To align with `Conditions`, we set scope of `when` expressions to the guarded `Task` and its dependent `Tasks`.

In [TEP-0059: Skipping Strategies][tep-0059], we proposed changing the scope of `when` expressions to the guarded `Task` only. This was implemented in #4085. We provided a feature flag, `scope-when-expressions-to-task`, to support migration. It defaulted to `false` for 9 months per our [Beta API compatibility policy][policy], meaning that we continued to guard the `Task` and its dependent `Tasks`. Then in #4580, we flipped the flag to `true` to guard the `Task` only by default.

In this change, we remove the `scope-when-expressions-to-task` flag and complete the migration.

Closes https://github.com/tektoncd/pipeline/issues/4461.

[tep-0007]: https://github.com/tektoncd/community/blob/main/teps/0007-conditions-beta.md
[tep-0059]: https://github.com/tektoncd/community/blob/main/teps/0059-skipping-strategies.md
[policy]: https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
The `scope-when-expressions-to-task` feature flag has been removed. 
```

### CODE REVIEW

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 1]
```diff
@@ -5083,7 +5072,7 @@ func TestReconcileWithWhenExpressionsScopedToTask(t *testing.T) {
 	}
 }
 
-func TestReconcileWithWhenExpressionsScopedToTaskWitResultRefs(t *testing.T) {
+func TestReconcileWithWhenExpressionsScopedToTaskWithResultRefs(t *testing.T) {
```
Comment (lbernick) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:5075]: should this test name also have "scopedToTask" removed?

**Review by lbernick — COMMENTED** [review]
/lgtm

File: pkg/reconciler/pipelinerun/pipelinerun_test.go (hunk 1 above)
Comment (jerop) [reply pkg/reconciler/pipelinerun/pipelinerun_test.go]: @lbernick agreed, updated the test name

**Review by dibyom — APPROVED** [review]
Nice cleanup!

**Comment by jerop** [comment]
```  zone: 'us-central1'>] finished with error: All cluster resources were brought up, but: only 2 nodes out of 3 have registered; this is likely due to Nodes failing to start correctly; check VM Serial logs for errors, try re-creating the cluster or contact support if that doesn't work. ```

/test pull-tekton-pipeline-integration-tests

**Pull request merged**